### DISCUSSION TITLE
Introspect PAT token

### DISCUSSION THREAD

**Discussion by satishviswanathan** [zitadel/zitadel#7672] [Q&A]
I have a PAT token generated. I'm planning to use this PAT token for a service to service communication. On the destination service i'm trying to do an introspect api call with this PAT token to get the roles so that I can check if the caller has the authorization to call me service as there are other end points on the destination service which I don't want to be exposed to the caller. However when I make the introspect call I'm getting the response that token is not active.

Please do let me know if this is a valid scenario , if so what I'm missing here.

**Comment by dakshitha** [comment]
Hi! Can you show me how you make the 1/ request to your API and 2/ request to the token introspection endpoint?
Also, FYI, this is a Python example of a use case similar to yours - https://zitadel.com/docs/examples/secure-api/python-flask
